src/renderer/components/EditRow.tsx

```diff
@@ -0,0 +1,41 @@
+import '../scss/RecentConnections.scss';
+import { Check, X } from 'react-bootstrap-icons';
+import { ConnectionModel } from 'db/Models';
+
+interface IProps {
+  value: ConnectionModel;
+  toggleReadOnly: () => void;
+  handleChange: (event) => void;
+}
+
+const EditRow = ({ value, toggleReadOnly, handleChange }: IProps) => {
+  if (value.connectionConfig.config === 'manual') {
+    return (
+      <tr key={value.id}>
+        <td>
+          <input
+            type="text"
+            required
+            defaultValue={value.nickname}
+            name="Nickname"
+          />
+        </td>
+        <td>{value.type}</td>
+        <td>{value.connectionConfig.address}</td>
+        <td>{value.connectionConfig.port}</td>
+        <td>{value.connectionConfig.username}</td>
+        <td>
+          <button type="button">
+            <Check onClick={(event) => handleChange(event)} />
```

Comment:
This is the event that's being fired - if you hover over 'event' in VSCode, its of type `React.MouseEvent<SVGElement, MouseEvent>`. This is definitely what the problem is - its some weird image thing. If you end up doing the useState solution, we could do something like this in `EditRow.tsx`:
```
const [newVal, setNewVal] = useState<string>('')
...
<input
  type="text"
  required
  value={newVal}
  onChange={(e) => {
    setNewVal(e.target.value);
  }}
/>
...
<Check onClick={() => handleSubmit(newValue)} 
```
with no event being passed, and then we would have handleSubmit(newValue) be something like this in `RecentConnections.tsx`:

```
const handleSubmit = (newValue) => {
  const model = connect.find((connection) => connection.id === edit);
    if (model !== undefined) {
      model.nickname = newValue;
      await window.connections.ipcRenderer.update(model);
    }
    setEdit(null);
}
```

This might not be exactly everything or right but could be a start